common/types.go

```diff
@@ -240,3 +242,63 @@ func (a *UnprefixedAddress) UnmarshalText(input []byte) error {
 func (a UnprefixedAddress) MarshalText() ([]byte, error) {
 	return []byte(hex.EncodeToString(a[:])), nil
 }
+
+// MixedcaseAddress retains the original string, which may or may not be
```

Comment:
Wouldn't it make more sense to verify the checksum when an address is received, and reject invalid ones?

Reply:
I don't think so, no.. At least I don't want to hardcode into the signer that it only ever can accept correct checksums. 
However, we could make it a flag, so that the signer defaults to rejecting invalid checksums, but can be configured to allow+warn.

Reply:
Is there a use-case for accepting invalid checksums? It seems dangerous to me.

Reply:
Well, for one 'backwards compatibility'. There may be any number of systems which needs things signed, but have not implemented checksums. I don't want to stop them from using the Clef

Reply:
If all the numbers and lowercase or uppercase, it should assume no checksum. Only mixed cases would be rejected, and imho no sane system should generate mixed case addresses that are not checksummed.